raster/src/main/scala/geotrellis/raster/ArrayMultibandTile.scala

```diff
@@ -428,4 +428,25 @@ class ArrayMultibandTile(_bands: Array[Tile]) extends MultibandTile with MacroMu
 
     new ArrayMultibandTile(newBands)
   }
+
+  override def equals(other: Any) = {
+    if (other == null) {
+      false
+    } else other match {
+      case that : ArrayMultibandTile =>
+        case class NotEqualException() extends Exception()
+        var result = true
+        try {
+          for (b <- 0 until bandCount) {
+            if (band(b) != that.band(b)) throw NotEqualException()
+          }
+        } catch {
+          case e : NotEqualException =>
```

Comment:
Note that `false && {println("hey"); true}` will not print and evaluate to false, right side is not executed if left is false